wgpu-core/src/command/clear.rs

```diff
@@ -349,4 +400,70 @@ pub(crate) fn collect_zero_buffer_copies_for_clear_texture(
             }
         }
     }
+
+    unsafe {
+        encoder.copy_buffer_to_texture(zero_buffer, dst_raw, zero_buffer_copy_regions.into_iter());
+    }
+}
+
+fn clear_texture_via_render_passes<A: hal::Api>(
+    dst_texture: &Texture<A>,
+    range: TextureInitRange,
+    clear_usage: hal::TextureUses,
+    encoder: &mut A::CommandEncoder,
+) -> Result<(), ClearError> {
+    let extent_base = wgt::Extent3d {
+        width: dst_texture.desc.size.width,
+        height: dst_texture.desc.size.height,
+        depth_or_array_layers: 1, // TODO: What about 3d textures? Only one slice a time, sure but how to select it?
+    };
+    let sample_count = dst_texture.desc.sample_count;
+    let is_3d_texture = dst_texture.desc.dimension == wgt::TextureDimension::D3;
+    for mip_level in range.mip_range {
+        let extent = extent_base.mip_level_size(mip_level, is_3d_texture);
+        for layer in range.layer_range.clone() {
+            let target = hal::Attachment {
+                view: dst_texture.get_clear_view(mip_level, layer),
+                usage: clear_usage,
+            };
+
+            if clear_usage == hal::TextureUses::DEPTH_STENCIL_WRITE {
+                unsafe {
+                    encoder.begin_render_pass(&hal::RenderPassDescriptor {
```

Comment:
nit: we could probably merge those branches a little closer, i.e.
`let (color_attachments, depth-stencil_attachment) = if ... else ...;`

Reply:
I think I already tried this back when I wrote this because the problem I'm getting now again looks awfully familiar: It's not easily possible since the color attachment array has variable length!

```
            let (color_attachments, depth_stencil_attachment) = if is_color {
                (
                    [hal::ColorAttachment {
                        target: hal::Attachment {
                            view: dst_texture.get_clear_view(mip_level, layer),
                            usage: hal::TextureUses::COLOR_TARGET,
                        },
                        resolve_target: None,
                        ops: hal::AttachmentOps::STORE,
                        clear_value: wgt::Color::TRANSPARENT,
                    }],
                    None,
                )
            } else {
                (
                    [],
                    Some(hal::DepthStencilAttachment {
                        target: hal::Attachment {
                            view: dst_texture.get_clear_view(mip_level, layer),
                            usage: hal::TextureUses::DEPTH_STENCIL_WRITE,
                        },
                        depth_ops: hal::AttachmentOps::STORE,
                        stencil_ops: hal::AttachmentOps::STORE,
                        clear_value: (0.0, 0),
                    }),
                )
            };
```

Reply:
You can always work around this by moving a variable up:
```rust
let color_attachments_tmp;
let (color_attachments, depth_stencil_attachment) = if is_color {
  color_attachments_tmp = [hal::ColorAttachment {
                        target: hal::Attachment {
                            view: dst_texture.get_clear_view(mip_level, layer),
                            usage: hal::TextureUses::COLOR_TARGET,
                        },
                        resolve_target: None,
                        ops: hal::AttachmentOps::STORE,
                        clear_value: wgt::Color::TRANSPARENT,
                    }];
                (
                    &color_attachments_tmp[..],
                    None,
                )
            } else {
                (
                    &[][..],
                    Some(hal::DepthStencilAttachment {
                        target: hal::Attachment {
                            view: dst_texture.get_clear_view(mip_level, layer),
                            usage: hal::TextureUses::DEPTH_STENCIL_WRITE,
                        },
                        depth_ops: hal::AttachmentOps::STORE,
                        stencil_ops: hal::AttachmentOps::STORE,
                        clear_value: (0.0, 0),
                    }),
                )
            };
```

Reply:
thx, makes sense! 👍